pkg/ccl/streamingccl/replicationtestutils/testutils.go

```diff
@@ -167,6 +168,28 @@ func (c *TenantStreamingClusters) WaitUntilHighWatermark(
 	})
 }
 
+// Similar to jobutils.WaitForJobToSucceed, but with a longer deadline.
+func waitForJobToSucceedWithin10Min(t testing.TB, db *sqlutils.SQLRunner, jobID jobspb.JobID) {
+	t.Helper()
+	testutils.SucceedsWithin(t, func() error {
+		var status string
+		var payloadBytes []byte
```

Comment:
I’m a bit worried that we have to wait 10 minutes for cutover to finish in a unit test. Did you get a sense on why we need this much time? Is it replication lag?

Reply:
this fails under stress and race, and in that environment anything can be starved for a while. this unit test is not a performance test so I'm ok with a few more minutes.
that said, if we can make the test run faster then that's always good.. I think right now it runs under 30 seconds (not under race), which is not too bad.

Reply:
Something we do in other places is set a higher timeout if we are under race using util.RaceEnabled. Perhaps something like:

```go
timeout := 2*time.Minutes
if util.RaceEnabled || skip.NightlyStress() {
    timeout = timeout * 5
}
```

Also, I don't mind too much, but if it were me I would probably export something like `WaitForJobToHaveStatus(testing.TB, *sqlutils.SQLRunner, jobspb.JobID, jobs.Status, time.Duration)` from jobutils and then call that here.

Reply:
I didn't want to add this to jobutils because I didn't want others to use it much. But doing this for stress/race only makes sense.

Anyway, I tried stressing it now and it doesn't fail with 2 minutes so I'll revert this for now and keep an eye on it in ci. This might have been another fix I did for running under race, before I gave up on it and just made it run under stress only. Or maybe it will fail again.. stressing locally is not super reliable.

BTW, if we think it's important to get this to run under race I can try that again, I have no intuition about what is expected or not expected to run well under race. But that can be a separate pr.

Hopefully this makes sense!